### PR TITLE
updating norms and adding learning goals

### PR DESCRIPTION
**Pull request by mariacommits**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: collaboration/learning_goals.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
 
-## Collective
+## Collective Goals
+- Review and practice data structures and algorithms.
+- Learn new programming techniques and tools.
+- Write clean, readable, and well-documented code.
+- Focus on solving problems efficiently and effectively.
+- Collaborate and provide constructive feedback to each other.
+- 
+## Individual Goals
```
Comment (ermishina) [inline collaboration/learning_goals.md:10]: Add a blank line before and after the ## Individual Goals heading to improve readability and formatting consistency

File: collaboration/learning_goals.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+
+### Micheal
```
Comment (ermishina) [inline collaboration/learning_goals.md:12]: The ### Michael header should be surrounded by blank lines to improve readability

File: collaboration/learning_goals.md [hunk 3]
```diff
@@ -1,5 +1,17 @@
 # Learning Goals
 
-## Collective
+## Collective Goals
+- Review and practice data structures and algorithms.
```
Comment (ermishina) [inline collaboration/learning_goals.md:4]: Please add a blank line before and after the list to comply with Markdown standards

File: collaboration/learning_goals.md [hunk 4]
```diff
@@ -1,5 +1,17 @@
...
+- Learn new programming techniques and tools.
+- Write clean, readable, and well-documented code.
+- Focus on solving problems efficiently and effectively.
+- Collaborate and provide constructive feedback to each other.
+- 
+## Individual Goals
+
+### Micheal
+- Strengthen understanding of data structures and algorithms.
```
Comment (ermishina) [inline collaboration/learning_goals.md:13]: Please add a blank line before and after the list to comply with Markdown standards

File: collaboration/README.md [hunk 5]
```diff
@@ -1,5 +1,3 @@
-# Collaboration
```
[the remark this answers is not in the document]
Comment (sky0-0glitch) [reply collaboration/README.md:1]: If you want to remove it, replace it with our group name. "ColabCrafter" has received the most votes so far, but not everyone has voted yet, so I am uncertain whether we will go with it.

**Review by sky0-0glitch — APPROVED** [review]

File: collaboration/learning_goals.md [hunk 6]
```diff
@@ -1,5 +1,17 @@
 # Learning Goals
 
-## Collective
+## Collective Goals
+- Review and practice data structures and algorithms.
+- Learn new programming techniques and tools.
```
Comment (sky0-0glitch) [inline collaboration/learning_goals.md:5]: add a blank line before the list to resolve the error

File: collaboration/learning_goals.md (hunk 1 above)
Comment (sky0-0glitch) [inline collaboration/learning_goals.md:10]: add blank line before and after your heading to resolve the Error

**Review by sky0-0glitch — CHANGES_REQUESTED** [review]

File: collaboration/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mariacommits) [reply collaboration/README.md:1]: I think we can put the group name in the main readme. I like "ColabCrafter"
